src/main/java/net/minecraftforge/dimension/DynamicDimensionManager.java

```diff
@@ -0,0 +1,221 @@
+	
+	{
+		Validate.notNull(loc, "Must provide valid ResourceLocation");
+		
+		RegistryKey<DimensionType> key = RegistryKey.func_240903_a_(Registry.field_239698_ad_, loc);
+		
+		Registry<DimensionType> reg = this.dynamicRegistries.func_230520_a_();
+		DimensionType type = reg.func_230516_a_(key);
+		
+		return type;
+	}
+	
+	/**
+	 * Retrieves the dimension under the specified {@link ResourceLocation}
+	 * @param loc the location of the dimension
+	 * @return the dimension
+	 */
+	@Nullable
+	public Dimension getDimension(ResourceLocation loc) 
+	{
+		Validate.notNull(loc, "Must provide valid ResourceLocation");
+		
+		RegistryKey<Dimension> registryKey = RegistryKey.func_240903_a_(Registry.field_239700_af_, loc);
+		
+		Dimension dimension = this.serverWorldInfo.func_230418_z_().func_236224_e_().func_230516_a_(registryKey);
+		
+		return dimension;
+	}
+	
+	/**
+	 * Retrieves worlds that are registered under the specified dimension type
+	 * @param server the minecraft server
+	 * @param loc the location of the world
+	 * @return the world
+	 */
+	@Nullable
+	public List<ServerWorld> getServerWorlds(MinecraftServer server, DimensionType type) 
+	{
+		Validate.notNull(server, "Must provide server when creating world");
+		Validate.notNull(type, "Must provide DimensionType");
+		
+		Registry<DimensionType> reg = this.dynamicRegistries.func_243612_b(Registry.field_239698_ad_);
+		
+		List<ServerWorld> worlds = new ArrayList<ServerWorld>();
+		
+		for(ServerWorld world : server.getWorlds()) 
+		{
+			if(reg.getKey(world.func_230315_m_()).equals(reg.getKey(type))) 
+			{
+				worlds.add(world);
+			}
+		}
+		
+		return worlds;
+	}
+	
+	/**
+	 * Retrieves the world instance of the dimension under the specified {@link ResourceLocation}
+	 * @param server the minecraft server
+	 * @param loc the location of the world
+	 * @return the world
+	 */
+	@Nullable
+	public World getServerWorld(MinecraftServer server, ResourceLocation loc) 
+	{
+		Validate.notNull(server, "Must provide server when creating world");
+		Validate.notNull(loc, "Must provide valid ResourceLocation");
+		
+		RegistryKey<World> registryKey = RegistryKey.func_240903_a_(Registry.field_239699_ae_, loc);
+		
+		return server.getWorld(registryKey);
+	}
+	
+	/**
+	 * Retrieves registered json noise settings from the registry
+	 * @param loc the resource lcoation of the dimension setting
+	 * @return
+	 */
+	@Nullable
+	public DimensionSettings getDimensionSettings(ResourceLocation loc)
+	{
+		Validate.notNull(loc, "Must provide valid ResourceLocation");
+		
+		RegistryKey<DimensionSettings> registryKey = RegistryKey.func_240903_a_(Registry.field_243549_ar, loc);
+		
+		DimensionSettings settings = this.dynamicRegistries.func_243612_b(Registry.field_243549_ar).func_230516_a_(registryKey);
+		
+		return settings;
+	}
+	
+	/**
+	 * Retrieves the Biome registry.
+	 * For default biome providers see {@link BiomeProviders}
+	 * @return the biome registry
+	 */
+	@Deprecated
```

Comment:
perhaps explain why this is deprecated?

Reply:
That's so modders don't get the idea of using this getter to get direct access to the BiomeRegistry but instead use my BiomeProviders class in case they want to create a standard BiomeProvider. Although I need it for my BiomeProviders class itself. Could have changed that to protected though.